Specs/Scene/Cesium3DTileBatchTableResourcesSpec.js

```diff
@@ -0,0 +1,459 @@
+/*global defineSuite*/
+defineSuite([
+        'Scene/Cesium3DTileBatchTableResources',
+        'Core/Cartesian3',
+        'Core/Color',
+        'Core/defaultValue',
+        'Core/HeadingPitchRange',
+        'Core/loadArrayBuffer',
+        'Renderer/ContextLimits',
+        'Scene/Cesium3DTileContentState',
+        'Scene/Cesium3DTileset',
+        'Specs/createScene',
+        'Specs/pollToPromise'
+    ], function(
+        Cesium3DTileBatchTableResources,
+        Cartesian3,
+        Color,
+        defaultValue,
+        HeadingPitchRange,
+        loadArrayBuffer,
+        ContextLimits,
+        Cesium3DTileContentState,
+        Cesium3DTileset,
+        createScene,
+        pollToPromise) {
+    "use strict";
+
+    var scene;
+    var centerLongitude = -1.31995;
+    var centerLatitude = 0.69871;
+
+    var withBatchTableUrl = './Data/Cesium3DTiles/Batched/BatchedWithBatchTable/';
+    var withoutBatchTableUrl = './Data/Cesium3DTiles/Batched/BatchedWithoutBatchTable/';
+    var batchLengthZeroUrl = './Data/Cesium3DTiles/Batched/BatchedNoBuildings/';
+
+    var result = new Color();
+
+    var fakeContentProvider = {
```

Comment:
Name this `mock` instead of `fake`.  https://github.com/AnalyticalGraphicsInc/cesium/blob/testing-guide/Documentation/Contributors/TestingGuide/README.md#mocks